docs/DEPLOYMENT.md

```diff
@@ -0,0 +1,410 @@
+## Setting up the ports webapp
+
+For testing and development purposes, the ports webapp can be deployed and populated locally. The main Django app is contained
+inside the `app` directory in the root of the project.
+
+###### Project structure
+```
+|-- app     (main django app)
+|-- config
+|-- docs
+|-- .gitignore
+|-- .gitmodules
+|-- .travis.yml
+|-- Dockerfile
+|-- env.sample
+|-- LICENSE
+|-- README.md
+```
+
+As mentioned, the `app` directory is a standalone django app. However, the app has been Dockerised and all the pain of
+setting up a local environment can be avoided when working with Docker. But in case you do not want to work with Docker,
+we will also walk through the standard process of setting up the app.
+
+## Getting the dependencies ready
+
+The external dependencies of the app are: PostgreSQL, Solr and Memcached. The search page(/search/) is completely derived from
+Solr. Memcached is optional, and can be avoided if you don't want to cache the results of stats and trac tickets.
+
+If you are new to these technologies, you may use the instructions provided below, otherwise you are free to install and setup these
+dependencies using your preferred mechanism.
+
+### PostgreSQL
+
+Download and install `Postgres.app` from [here](https://postgresapp.com/downloads.html).
+
+After this, connect as root:
+```bash
+su - postgres
+```
+
+Create your user(replace username and password with your preferred credentials):
+```bash
+CREATE USER username PASSWORD 'passowrd';
+```
+
+Create a new database;
+
+```bash
+CREATE DATABASE webapp;
+```
+
+Grant permissions to your user on this database.
+
+```bash 
+GRANT ALL PRIVILEGES ON DATABASE webapp TO username;
+```
+
+That's it for PostgreSQL, our database for the app is ready.
+
+### Solr
+
+To get the required version of Solr working we need [openjdk8](https://openjdk.java.net/install/). On a mac, you may use 
+[MacPorts to install openjdk8](https://ports.macports.org/port/openjdk8/). New versions of openjdk may raise problems,
+we recommend openjdk8.
+
+Once opendjk8 has been installed, we can proceed to downloading and launching Solr. We would need 
+[Solr 6.6.6](https://archive.apache.org/dist/lucene/solr/6.6.6/). The below steps are self explanatory:
```

Comment:
I see that we provide `apache-solr8` as a package as well. If we keep mentioning macports, we could probably mention this port as well, unless it's not suitable for some reason.

Just a question. I see that version 8.9.0 is out. This one points to 6.6.6. Is there some reason to use outdated version (other than the fact that these docs were written long time ago)?

Reply:
`6.6.6` is the latest django-haystack supports. I had this discussion with Amar last year during the development phase and he said that is fine.

Reply:
Sure, that's fine. It's just probably helpful to explicitly mention the reason for sticking with a particular version when that's the case. Else one might decide to update everything one day and waste time debugging why things don't work.